src/sentry/static/sentry/app/components/modals/inviteMembersModal/inviteRowControl.tsx

```diff
@@ -24,94 +27,182 @@ type Props = {
+            disabled={disabled}
+            placeholder={t('Enter one or more emails')}
+            inputValue={this.state.inputValue}
+            value={emails}
+            components={{
+              MultiValue: (props: MultiValueProps<SelectOption>) =>
+                ValueComponent(props, inviteStatus),
+            }}
+            options={mapToOptions(emails)}
+            onBlur={(e: React.ChangeEvent<HTMLInputElement>) =>
+              e.target.value &&
+              onChangeEmails([
+                ...mapToOptions(emails),
+                {label: e.target.value, value: e.target.value},
+              ])
+            }
+            styles={{
+              multiValue: (
+                provided: React.CSSProperties,
+                {data}: MultiValueProps<SelectOption>
+              ) => {
+                const status = inviteStatus[data.value];
+                return {
+                  ...provided,
+                  ...(status && status.error
+                    ? {
+                        color: theme.red300,
+                        border: `1px solid ${theme.red300}`,
+                        backgroundColor: theme.red100,
+                      }
+                    : {}),
+                };
+              },
+              multiValueLabel: (
+                provided: React.CSSProperties,
+                {data}: MultiValueProps<SelectOption>
+              ) => {
+                const status = inviteStatus[data.value];
+                return {
+                  ...provided,
+                  pointerEvents: 'all',
+                  ...(status && status.error ? {color: theme.red300} : {}),
+                };
+              },
+              multiValueRemove: (
+                provided: React.CSSProperties,
+                {data}: MultiValueProps<SelectOption>
+              ) => {
+                const status = inviteStatus[data.value];
+                return {
+                  ...provided,
+                  ...(status && status.error
+                    ? {
+                        borderLeft: `1px solid ${theme.red300}`,
+                        ':hover': {backgroundColor: theme.red100, color: theme.red300},
+                      }
+                    : {}),
+                };
+              },
+            }}
+            onInputChange={this.handleInputChange}
+            onKeyDown={this.handleKeyDown}
+            onBlurResetsInput={false}
+            onCloseResetsInput={false}
+            onChange={onChangeEmails}
+            multiple
+            creatable
+            clearable
+            menuIsOpen={false}
+          />
+        </div>
+        <div>
+          <RoleSelectControl
+            data-test-id="select-role"
+            disabled={disabled}
+            value={role}
+            roles={roleOptions}
+            disableUnallowed={roleDisabledUnallowed}
+            onChange={onChangeRole}
+          />
+        </div>
+        <div>
```

Comment:
Are these extra divs necessary for layout? Or can we remove them?

Reply:
Looks like they aren't needed. I'll remove them.